lib/Backend/GlobOptBailOut.cpp

```diff
@@ -15,52 +15,258 @@ GlobOpt::CaptureValue(BasicBlock *block, StackSym * stackSym, Value * value, Bai
+
+        tempBv->Set(Js::Constants::InvalidSymID);
+
+        FOREACH_BITSET_IN_SPARSEBV(symId, tempBv)
         {
-            continue;
+            Sym * sym = hasConstValue ? iterConst.Data().Key() : nullptr;
+            Value * val = nullptr;
+            HashBucket<Sym *, Value *> * symIdBucket = nullptr;
+            bool symIdHandled = false;
+
+            // copy unchanged sym to new capturedValues
+            while (sym && sym->m_id < symId)
+            {
+                Assert(sym->IsStackSym());
+                if (!sym->AsStackSym()->HasArgSlotNum() || block->globOptData.capturedArgs->Test(sym->m_id))
+                {
+                    bailOutConstValuesIter.InsertNodeBefore(this->func->m_alloc, sym->AsStackSym(), iterConst.Data().Value());
+                }
+
+                hasConstValue = iterConst.Next();
+                sym = hasConstValue ? iterConst.Data().Key() : nullptr;
+            }
+            if (sym && sym->m_id == symId)
+            {
+                hasConstValue = iterConst.Next();
+            }
+            if (symId != Js::Constants::InvalidSymID)
+            {
+                // recapture changed constant sym
+
+                symIdBucket = block->globOptData.symToValueMap->GetBucket(symId);
+                if (symIdBucket == nullptr)
+                {
+                    continue;
+                }
+
+                sym = symIdBucket->value;
+                Assert(sym->IsStackSym() && (sym->AsStackSym()->HasByteCodeRegSlot() || sym->AsStackSym()->HasArgSlotNum()));
+
+                val =  block->globOptData.capturedArgs->Test(symId) ? FindValue(sym) : symIdBucket->element;
+                ValueInfo* valueInfo = val->GetValueInfo();
+
+                if (valueInfo->GetSymStore() != nullptr)
+                {
+                    int32 intConstValue;
+                    BailoutConstantValue constValue;
+
+                    if (valueInfo->TryGetIntConstantValue(&intConstValue))
+                    {
+                        constValue.InitIntConstValue(intConstValue);
+                        bailOutConstValuesIter.InsertNodeBefore(this->func->m_alloc, sym->AsStackSym(), constValue);
+                        symIdHandled = true;
+                    }
+                    else if(valueInfo->IsVarConstant())
+                    {
+                        constValue.InitVarConstValue(valueInfo->AsVarConstant()->VarValue());
+                        bailOutConstValuesIter.InsertNodeBefore(this->func->m_alloc, sym->AsStackSym(), constValue);
+                        symIdHandled = true;
+                    }
+                }
+                else if (!valueInfo->HasIntConstantValue())
+                {
+                    continue;
+                }
+            }
+
+            if (symIdHandled)
+            {
+                continue;
+            }
+
+            sym = hasCopyPropSym ? iterCopyPropSym.Data().Key() : nullptr;
+
+            while (true)
```

Comment:
The structure is similar, but this loop need to handle more things because the change in copy sym will also trigger recapture path before reaching changed symId.

---
In reply to: [68347583](https://github.com/Microsoft/ChakraCore/pull/1152#discussion_r68347583) [](ancestors = 68347583)